CONTRIBUTING.md

```diff
@@ -0,0 +1,318 @@
+They also have to have mechanisms to prevent them to act when the machine does not need the fix. The Ansible's language works in such a way that this behavior is built-in, however, for the other types, the remediations have to implement it explicitly. 
+Remediations also carry metadata that should be present at the beginning of the files, that is how it looks like and what it means:
+```yml
+# platform = multi_platform_all
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+```
+Field | Description | Accepted values
+----------:| :------------ |:-:
+platform   | On which platforms it can be applied | multi_platform_all, multi_platform_oval, multi_platform_oval, multi_platform_fedora, multi_platform_debian, multi_platform_ubuntu, multi_platform_wrlinux, multi_platform_rhel, multi_platform_openstack, multi_platform_opensuse, multi_platform_rhev, multi_platform_sle, Red Hat Enterprise Linux 7, **TODO**
+reboot 	   | Whether or not a reboot is necessary after the fix | true, false
+strategy   | **TODO** | enable, disable, unknown, restrict, configure, patch
+complexity | **TODO** | low, medium, high
+disruption | **TODO** | low, medium, high
+
+#### Ansible
+
+Ansible remediations are stored as yml files in directory */template/static/ansible* under the targeted platform. They are meant to be executed by Ansible itself when requested by openscap, so they are
+written using [Ansible's own language](http://docs.ansible.com/ansible/latest/intro.html) with the following exceptions:
+
+- The remediation content must be only the *tasks* section of what would be a playbook
+- The *tags* section must be present in each task as shown in the example, it'll be replaced during the building process
+- Notifications and handlers are not supported
+
+Here is an example of a Ansible remediation that ensures the SELinux is enabled in grub:
+```yml
+# platform = multi_platform_rhel,multi_platform_fedora
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+- name: Ensure SELinux Not Disabled in /etc/default/grub
+  replace:
+    dest: /etc/default/grub
+    regexp: selinux=0
+  tags:
+    @ANSIBLE_TAGS@
+```
+
+#### Bash
+
+Bash remediations are stored as shell script files in directory */template/static/bash* under the targeted platform. They have to make sure they are not trying to fix a system which is not broken.
+
+Following, you can see an example of a bash remediation that sets the maximum number of days a password may be used:
+``` sh
+# platform = Red Hat Enterprise Linux 7
+. /usr/share/scap-security-guide/remediation_functions
+populate var_accounts_maximum_age_login_defs
+
+grep -q ^PASS_MAX_DAYS /etc/login.defs && \
+  sed -i "s/PASS_MAX_DAYS.*/PASS_MAX_DAYS     $var_accounts_maximum_age_login_defs/g" /etc/login.defs
+if ! [ $? -eq 0 ]; then
+    echo "PASS_MAX_DAYS      $var_accounts_maximum_age_login_defs" >> /etc/login.defs
+fi
+```
```

Comment:
I would prefer to show case a remediation that uses `replace_or_append` function. Use of `grep` and `sed` for the example above is very error prone.
And it would also be a good place to mention bash remediation functions.